src/app/public/modules/dock/dock.service.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  ComponentRef,
+  ElementRef,
+  Injectable,
+  Renderer2,
+  RendererFactory2
+} from '@angular/core';
+
+import {
+  MutationObserverService,
+  SkyDynamicComponentService
+} from '@skyux/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import 'rxjs/add/observable/fromEvent';
+
+import 'rxjs/add/operator/debounceTime';
+
+import {
+  SkyDockComponent
+} from './dock.component';
+
+import {
+  SkyDockingOptions
+} from './docking-options';
+
+const ATTR_STACK_ORDER = 'data-sky-dock-stack-order';
+
+/**
+ * This service docks elements to specific areas on the page.
+ */
+@Injectable()
+export class SkyDockService {
+
+  private get bottomDockElement(): HTMLElement {
+    return this.bottomDockRef.instance.elementRef.nativeElement as HTMLElement;
+  }
+
+  private bottomDockHeight: number;
+
+  private bottomDockObserver: MutationObserver;
+
+  private bottomDockRef: ComponentRef<SkyDockComponent>;
+
+  private bottomDockStyleElement: HTMLStyleElement;
+
+  private ngUnsubscribe: Subject<void>;
+
+  private renderer: Renderer2;
+
+  constructor(
+    private mutationService: MutationObserverService,
+    private dynamicComponentService: SkyDynamicComponentService,
+    rendererFactory: RendererFactory2
+  ) {
+    this.renderer = rendererFactory.createRenderer(undefined, undefined);
+  }
+
+  /**
+   * Docks an element to the bottom of the page.
+   * @param elementRef The element to dock.
+   * @param options Options that affect the docking action.
+   */
+  public dockToBottom(elementRef: ElementRef, options: SkyDockingOptions = {}): void {
```

Comment:
Good observation. Ignoring the `private` methods (those can easily be changed without breaking changes), the reason I went with `dockToBottom` and `removeFromBottom` is for the sake of simplicity for the API and our own internal implementation.
1. If the consumer wants to add an element to the bottom dock, all they have to do is call `dockToBottom(elementRef)`. They don't have to worry about messing with a config for every docking action. Also, the method name is implicitly understandable and communicates what it will do without the consumer providing additional configuration. `dockToBottom()` communicates better than `dock()` or `addToDock()`, especially since the "dock" element is invisible to the consumer (if the dock _was_ visible to the consumer, it would be immediately obvious where the element would be docked, so the word "bottom" would be unnecessary).
2. Having a separate method for each location will allow us more flexibility if each location needs its own specific configuration or parameters (maybe the left-hand dock needs specific options that wouldn't apply to the bottom dock, for example).
3. The removal action gets more complicated if we don't specify the location in the method call. For example, if we renamed the `removeFromBottom(elementRef)` method to `remove(elementRef)`, the internal implementation would have to do a query into every dock element to see if it exists. Obviously this isn't a huge deal, but it adds complexity.

Reply:
(Happy to hear you out if you think my reasons aren't valid.)

Reply:
All good points. You've convinced me!